### PR TITLE
TypedData_Get_Struct unwraps data for a parent type

### PR DESCRIPTION
**Pull request by kaorimatz**

### CODE REVIEW

**Comment by ko1** [comment]
DO NOT USE Parent type. It is obsolete.
This is why we note "You have to fill reserved and parent with 0." in https://github.com/ruby/ruby/blob/trunk/doc/extension.rdoc

**Comment by eregon** [comment]
@ko1 Good to know.
I guess the documentation could be clearer about parent being obsolete.
Is there a replacement for it or is it just no longer needed?

BTW, does an inline struct declaration like https://github.com/ruby/spec/pull/416/files#diff-7ad8ada0216a72e8cc13d96a1cebadaeR27 always fill remaining fields with `0` in C?

**Comment by ko1** [comment]
>  Is there a replacement for it or is it just no longer needed?

I believe this feature is not needed. This field was introduced by accident (by nobu).

> BTW, does an inline struct declaration like https://github.com/ruby/spec/pull/416/files#diff-7ad8ada0216a72e8cc13d96a1cebadaeR27 always fill remaining fields with 0 in C?

Yes.

File: optional/capi/ext/typed_data_spec.c [hunk 1]
```diff
@@ -30,6 +54,7 @@ static const rb_data_type_t sample_typed_wrapped_struct_data_type = {
     sample_typed_wrapped_struct_free,
     sample_typed_wrapped_struct_memsize,
   },
+  &sample_typed_wrapped_struct_parent_data_type,
```
Comment (eregon) [inline optional/capi/ext/typed_data_spec.c]: @ko1 Removing this line makes the new spec fail:
```
TypeError: wrong argument type sample_typed_wrapped_struct (expected sample_typed_wrapped_struct_parent)
```

Is the code in `sws_typed_get_struct_parent_type` not supported?
Is it possible to express some kind of inheritance with TypedData then?

**Pull request merged**